Feat[#123]: 백오피스 컨텐츠 추가 구현

likerhythm opened the pull request:
## 확인 사항
- [x] 💯 테스트는 잘 통과했나요?
- [x] 🏗️ 빌드는 성공했나요?
- [x] 🧹 불필요한 코드는 제거했나요?
- [x] 💭 이슈는 등록했나요?
- [x] 🏷️ 라벨은 등록했나요?

## 작업 내용
백오피스에서 컨텐츠를 추가하는 API를 구현했습니다.
컨텐츠 추가 시 컨텐츠의 임베딩 값을 계산해야 하기 때문에 FastAPI 서버의 api도 구현해 놓았습니다.

## 시퀀스 다이어 그램
<img width="1506" height="751" alt="image" src="https://github.com/user-attachments/assets/77343e25-595d-43b9-865e-f189ee320059" />

Closes #123

likerhythm committed:
feat[#123]: 컨텐츠 추가 - 컨트롤러 계층 구현

likerhythm committed:
fix[#123]: 컨텐츠 추가 api 리턴값 수정

likerhythm committed:
feat[#123]: 컨텐츠 추가 - dto 구현

likerhythm committed:
feat[#123]: 컨텐츠 추가 - 서비스 계층 구현

likerhythm committed:
refactor[#123]: AdminAddContentRequestDto 불필요 빌드 삭제

likerhythm committed:
feat[#123]: ContentMapper에 dto 매퍼 추가

likerhythm committed:
feat[#123]: 컨텐츠 추가 - 서비스 계층 구현

likerhythm committed:
feat[#123]: 컨텐츠 추가 - 리포지토리 계층 구현

likerhythm committed:
feat[#123]: 존재하지 않는 나라인 경우 예외 처리

likerhythm committed:
feat[#123]: genres, actors, director 필드 추가

likerhythm committed:
fix[#123]: ContentMapper opendate 파싱 오류 해결

likerhythm committed:
feat[#123]: Content updateEmbedding 메서드 추가

likerhythm committed:
feat[#123]: MetaInfo 에러코드 추가

likerhythm committed:
feat[#123]: fastapi 서버로부터 벡터를 받아오는 기능 추가

likerhythm committed:
feat[#123]: MetaInfoRepository 이름으로 조회하는 쿼리 추가

likerhythm committed:
feat[#123]: 컨텐츠 추가 - 서비스 계층 구현

likerhythm committed:
test[#123]: AdminContentService 테스트 구현

likerhythm committed:
feat[#123]: MetaInfoRepository를 querydsl로 대체

likerhythm committed:
fix[#123]: 실수로 지운 import 복구

likerhythm committed:
feat[#123]: User role 검증 로직 추가

likerhythm committed:
feat[#123]: 어드민 Fixture 구현

likerhythm committed:
test[#123]: 컨텐츠 추가 - 권한 테스트

likerhythm committed:
fix[#123]: 잘못된 메서드 호출 수정

likerhythm committed:
Merge branch 'develop' into feat/backoffice-content-cud

likerhythm committed:
refactor[#123]: 수정사항 반영

sangyunpark99 commented on src/main/java/org/highfive/backend/content/controller/AdminContentController.java:
```diff
@@ -0,0 +1,26 @@
+
+    @PostMapping("/content")
+    public Response<AdminAddContentResponseDto> adminAddContent(@Valid @RequestBody AdminAddContentRequestDto request,
+                                                                @AuthenticationPrincipal User user) {
```
final 키워드 붙이는게 어떨까요?

likerhythm committed:
fix[#123]: 홈 컨텐츠 조회 주소 수정

sangyunpark99 commented on src/main/java/org/highfive/backend/content/controller/HomeContentController.java:
```diff
@@ -15,7 +15,7 @@ public class HomeContentController {
     private final HomeContentService homeContentService;
 
-    @GetMapping("/content/recommend")
+    @GetMapping("/content/home")
```
url을 변경하니 더 명확하네요!

sangyunpark99 commented on src/main/java/org/highfive/backend/content/dto/response/AdminAddContentResponseDto.java:
```diff
@@ -0,0 +1,15 @@
+@AllArgsConstructor
+public class AdminAddContentResponseDto {
+
+    private Long contentId;
```
생성자를 사용하지 않고, 굳이 Settter를 사용한 이유가 뭘까요? 궁금합니다.
dto를 record 자료형으로 하는게 어떨까요?

likerhythm replied:
습관적으로 Setter랑 class를 사용했네요 record로 바꾸겠습니다~!

sangyunpark99 commented on src/main/java/org/highfive/backend/content/repository/querydsl/QueryDslMetaInfoRepository.java:
```diff
@@ -0,0 +1,25 @@
+    public MetaInfo findByNameAndType(String name, MetaType type) {
+        return queryFactory
+                .selectFrom(metaInfo)
+                .where(metaInfo.name.eq(name), metaInfo.type.eq(type))
```
JPA로도 충분히 로직을 구현할 수 있다고 생각합니다!

sangyunpark99 commented on src/main/java/org/highfive/backend/content/service/AdminContentService.java:
```diff
@@ -0,0 +1,86 @@
+            throw new BusinessException(GlobalErrorCode.ACCESS_DENIED);
+        }
+
+        Content content = ContentMapper.fromAdminAddContentRequestDto(request);
```
final 키워드 붙이는게 어떨까요?

sangyunpark99 commented on src/main/java/org/highfive/backend/content/service/OnboardingService.java:
```diff
@@ -52,7 +54,7 @@ public class OnboardingService {
     private final UserRepository userRepository;
     private final ContentRepository contentRepository;
-    private final MetaInfoRepository metaInfoRepository;
+    private final QueryDslMetaInfoRepository queryDslMetaInfoRepository;
```
MeataInfoQueryRepository로 이름을 명명하는게 어떨까요?

sangyunpark99 commented on src/main/java/org/highfive/backend/global/client/fastapi/dto/response/FastApiVectorFromGenresDto.java:
```diff
@@ -0,0 +1,13 @@
+@AllArgsConstructor
+public class FastApiVectorFromGenresDto {
+
+    String vector;
```
record를 사용하는게 어떨까요?

sangyunpark99 reviewed (changes requested):
수고하셨습니다

likerhythm committed:
chore[#123]: pr 리뷰 반영

- QueryDslMetaInfoRepository 삭제 후 JPA Query 사용하도록 변경
- final 키워드 추가
- dto에는 record 사용하도록 수정

sangyunpark99 reviewed (approved):
수고하셨습니다!
